folksonomy/api.py

```diff
@@ -352,23 +359,45 @@ async def product_tag_update(response: Response,
     """
 
     check_owner_user(user, product_tag.owner, allow_anonymous=False)
+
     try:
-        timing = await db_exec("""
-UPDATE folksonomy SET v = %s, version = %s, editor = %s, comment = %s
-    WHERE product = %s AND owner = %s AND k = %s
-    """, (product_tag.v, product_tag.version, user, product_tag.comment,
-            product_tag.product, product_tag.owner, product_tag.k.lower()))
+        await db_exec("""
+    SELECT version FROM folksonomy WHERE product = %s AND owner = %s AND k = %s
```

Comment:
I prefer to apply the UPDATE first, then in case it fails we may do a SELECT to get the last version. In most cases, only the UPDATE will be executed (and atomically).

I'm not in favor of returning the last version number without the whole object associated because this allows to send the right version number in a later put without getting the missed version of the object in between.

Reply:
@cquest I had brought in version checking in the API code. @alexgarel pointed out to me, that it was already implemented. So, there are no changes in the API code from my side. It's the same code as the main branch.

Reply:
Sorry, I don't understand. The current code in master does an UPDATE and nothing else.
The UPDATE will do the version check at postgresql level: https://github.com/openfoodfacts/folksonomy_api/blob/main/db/db_setup.sql#L22 and will return an error if the version does not match.

Your commit replaces it with a SELECT first, to check the version then an UPDATE.

Reply:
I don't think so. The "files changed" column only shows some spaces changing between my api.py and the current api.py code.